bin/cheribsdtest/cheribsdtest_registers.c

```diff
@@ -166,11 +166,6 @@ check_initreg_code(void * __capability c)
 		cheribsdtest_failure_errx("perms %jx (system_regs present)", v);
 
 	expect = CHERI_PERMS_SWALL & ~CHERI_PERM_SW_VMEM;
-#ifdef CHERIBSD_C18N_TESTS
-#ifndef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
-	expect &= ~CHERI_PERM_SYSCALL;
-#endif
-#endif
```

Comment:
Test removal should be a separate commit so it can be reverted after the release is out. That being said, this argues for changing the kernel's enforcement not rtld's implementation.